### PR TITLE
Refactor connection string parsing

### PR DESCRIPTION
**Pull request by nr-ahemsath**
### Description

This PR improves the robustness of the configuration classes for the unbounded integration tests, by replacing positional parsing of connection strings to extract values needed for assertions (e.g. server host) with use of `System.Data.Common.DbConnectionStringBuilder` (thanks to @bgrainger for the suggestion to use this class).

### Testing

I verified that the unbounded integration tests all pass for me locally with these changes.

### Changelog

N/A

### CODE REVIEW

**Review by bgrainger — APPROVED** [review]

**Comment by alanwest** [comment]
Sorry futzed with things on mobile device and accidentally hit close... Don't mind me :)

File: tests/Agent/IntegrationTests/Shared/MsSqlOdbcConfiguration.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 
 using System;
+using System.Collections.Generic;
```
Comment (jaffinito) [inline tests/Agent/IntegrationTests/Shared/MsSqlOdbcConfiguration.cs:6]: This can be removed since its not used.

**Review by jaffinito — APPROVED** [review]
Other than the using comment, this looks great!

File: tests/Agent/IntegrationTests/Shared/MsSqlOdbcConfiguration.cs (hunk 1 above)
Comment (nr-ahemsath) [reply tests/Agent/IntegrationTests/Shared/MsSqlOdbcConfiguration.cs]: Thanks for noticing, my last push removed that.

**Pull request merged**